src/main/java/org/apache/sling/feature/builder/BuilderUtil.java

```diff
@@ -131,21 +130,22 @@ static void mergeVariables(Map<String,String> target, Map<String,String> source,
      * @throws IllegalStateException If bundles can't be merged, for example if no
      *                               override is specified for a clash.
      */
-    static void mergeBundles(final Bundles target,
-        final Bundles source,
+    static void mergeArtifacts(final Artifacts target,
+        final Artifacts source,
         final Feature sourceFeature,
             final List<ArtifactId> artifactOverrides,
         final String originKey) {
 
         for (final Artifact artifactFromSource : source) {
 
             // set of artifacts in target, matching the artifact from source
-            final Set<Artifact> allExistingInTarget = new HashSet<>();
+            final Set<Artifact> allExistingInTarget = new LinkedHashSet<>();
```

Comment:
Would be good to add a comment to explain why a LinkedHashSet is used?

Reply:
I added some comments